src/lib.rs

```diff
@@ -525,10 +527,37 @@ fn run_single_on_query_match_callback<'a, 'b, 'c>(
     }
 }
 
-// pub struct RunForSliceStatus {
-//     pub violations: Vec<ViolationWithContext>,
-//     pub from_file_run_context_instance_provider: Box<dyn FromFileRunContextInstanceProvider>,
-// }
+#[self_referencing]
+pub struct PerConfigContext {
```

Comment:
Use a self-referencing struct to encapsulate the instantiated rules + the aggregated queries (which want to contain a reference to the instantiated rules)